cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,161 @@
+from django.urls import reverse
+from rest_framework.exceptions import ErrorDetail
+from rest_framework.test import APITestCase, APIClient
+from rest_framework import status
+from cinema.models import Movie, Genre, Actor
+from django.contrib.auth import get_user_model
+from rest_framework.authtoken.models import Token
+from io import BytesIO
+from PIL import Image
+
+User = get_user_model()
+
+
+def create_genre(name: str) -> Genre:
+    return Genre.objects.create(name=name)
+
+
+def create_actor(first_name: str, last_name: str) -> Actor:
+    return Actor.objects.create(first_name=first_name, last_name=last_name)
+
+
+def create_movie(title: str, genres=None, actors=None) -> Movie:
+    movie = Movie.objects.create(
+        title=title, description="Sample description", duration=120
+    )
+    if genres:
+        movie.genres.set(genres)
+    if actors:
+        movie.actors.set(actors)
+    return movie
+
+
+def generate_image_file():
+    image = Image.new("RGB", (100, 100))
+    byte_arr = BytesIO()
+    image.save(byte_arr, format="PNG")
+    byte_arr.seek(0)
+    return byte_arr
+
+
+class MovieViewSetTests(APITestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = User.objects.create_user(
+            email="testuser@example.com", password="testpass"
+        )
+        self.admin_user = User.objects.create_superuser(
+            email="adminuser@example.com", password="adminpass"
+        )
+        self.token = Token.objects.create(user=self.user)
+        self.admin_token = Token.objects.create(user=self.admin_user)
+        self.client.credentials(HTTP_AUTHORIZATION=f"Token {self.token.key}")
+
+        self.genre1 = create_genre("Action")
+        self.genre2 = create_genre("Drama")
+
+        self.actor1 = create_actor("John", "Doe")
+        self.actor2 = create_actor("Jane", "Smith")
+
+        self.movie1 = create_movie(
+            "Movie One", genres=[self.genre1], actors=[self.actor1]
+        )
+        self.movie2 = create_movie(
+            "Movie Two", genres=[self.genre2], actors=[self.actor2]
+        )
+
+    def test_list_movies(self):
+        url = reverse("cinema:movie-list")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+
```

Comment:
Your test cases would benefit from more detailed assertion messages. For example, in your `assertEqual` calls, you can add a third argument that will be printed if the assertion fails. This will make it easier to debug failing tests. For example: `self.assertEqual(len(response.data), 2, 'Expected to find 2 movies in the response')`